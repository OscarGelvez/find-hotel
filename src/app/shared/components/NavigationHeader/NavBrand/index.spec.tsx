import '@testing-library/jest-dom';

import { cleanup, render } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import { NavBrand } from './index';

afterEach(cleanup);

it('rederizar el componente NavBrand sin errores', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <NavBrand imgSrc={''} text={''} />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renderiza el componente NavBrand cuando se envía una imagen', () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <NavBrand
        imgSrc={'htpp://image.com/img.png'}
        text={'Text or logo header'}
      />
    </BrowserRouter>
  );
  expect(getByTestId('div-brand')).toContainElement(getByTestId('link-brand'));
  expect(getByTestId('div-brand')).toContainElement(getByTestId('image-brand'));
});

it('renderiza el componente NavBrand cuando se envía un texto', () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <NavBrand imgSrc={''} text={'Text or logo header'} />
    </BrowserRouter>
  );
  expect(getByTestId('div-brand')).toContainElement(getByTestId('link-brand'));
  expect(getByTestId('div-brand')).toContainElement(getByTestId('text-brand'));
});

it('Compara snapshot del componente NavBrand cuando se envía imagen', () => {
  const element = renderer
    .create(
      <BrowserRouter>
        <NavBrand
          imgSrc={'htpp://image.com/img.png'}
          text={'Text or logo header'}
        />
      </BrowserRouter>
    )
    .toJSON();
  expect(element).toMatchSnapshot();
});

it('Compara snapshot del componente NavBrand cuando se envía texto', () => {
  const element = renderer
    .create(
      <BrowserRouter>
        <NavBrand imgSrc={''} text={'Text or logo header'} />
      </BrowserRouter>
    )
    .toJSON();
  expect(element).toMatchSnapshot();
});
