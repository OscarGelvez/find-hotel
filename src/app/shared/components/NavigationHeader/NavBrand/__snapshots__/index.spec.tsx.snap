// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compara snapshot del componente NavBrand cuando se envía imagen 1`] = `
<div
  className="sc-AxirZ imQSvU"
  data-testid="div-brand"
>
  <a
    className="sc-AxjAm jqjyPd"
    data-testid="link-brand"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="Text or logo header"
      className="sc-AxiKw iunsIC"
      data-testid="image-brand"
      src="htpp://image.com/img.png"
    />
  </a>
</div>
`;

exports[`Compara snapshot del componente NavBrand cuando se envía texto 1`] = `
<div
  className="sc-AxirZ imQSvU"
  data-testid="div-brand"
>
  <a
    className="sc-AxjAm jqjyPd"
    data-testid="link-brand"
    href="/"
    onClick={[Function]}
  >
    <span
      className="sc-AxhCb cqzvYV"
      data-testid="text-brand"
    >
      Text or logo header
    </span>
  </a>
</div>
`;
